lib/internal/fs/promises.js

```diff
@@ -610,6 +614,9 @@ async function readv(handle, buffers, position) {
 }
 
 async function write(handle, buffer, offsetOrOptions, length, position) {
+  if (isArrayBuffer(buffer)) {
+    buffer = new Uint8Array(buffer);
+  }
   if (buffer?.byteLength === 0)
     return { __proto__: null, bytesWritten: 0, buffer };
```

Comment:
I think it makes more sense to keep the shortcut
```suggestion
  if (buffer?.byteLength === 0)
    return { __proto__: null, bytesWritten: 0, buffer };

  if (isArrayBuffer(buffer)) {
    buffer = new Uint8Array(buffer);
  }
```

Reply:
I've just tested passing an ArrayBuffer directly to the C++ binding, and it causes the program to crash.

```../../src/node_file.cc:2026:void node::fs::WriteBuffer(const FunctionCallbackInfo<v8::Value> &): Assertion `Buffer::HasInstance(args[1])' failed.
 1: 0x102fcc05c node::Abort() [/projects/node/node/out/Release/node]
 2: 0x102fcbe90 node::AppendExceptionLine(node::Environment*, v8::Local<v8::Value>, v8::Local<v8::Message>, node::ErrorHandlingMode) [/projects/node/node/out/Release/node]
 3: 0x102fdfffc node::fs::WriteBuffer(v8::FunctionCallbackInfo<v8::Value> const&) [/projects/node/node/out/Release/node]
 ... 30 more lines in stack trace
--- CRASHED (Signal: 6) ---```